contracts/MultiRoundCheckout.sol

```diff
@@ -8,16 +8,35 @@ import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/IERC20PermitU
 import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
 import "./IVotable.sol";
 import "./IDAIPermit.sol";
+import "./IAllo.sol";
 
 error VotesNotEqualRoundsLength();
 error AmountsNotEqualRoundsLength();
 error ExcessAmountSent();
+error INVALID_INPUT();
 
 contract MultiRoundCheckout is OwnableUpgradeable, PausableUpgradeable, ReentrancyGuardUpgradeable {
-    function initialize() public initializer {
+    modifier validateNativeBalance() {
+        uint256 initialBalance = address(this).balance;
+        _;
+        if (address(this).balance != initialBalance - msg.value) {
+            revert ExcessAmountSent();
+        }
+    }
+
+    modifier validateErc20Balance(address token) {
+        uint256 initialBalance = IERC20Upgradeable(token).balanceOf(address(this));
+        _;
+        if (IERC20Upgradeable(token).balanceOf(address(this)) != initialBalance) {
+            revert ExcessAmountSent();
+        }
+    }
+
```

Comment:
shall we add the constructor with `_disableInitializers();` too as proposed by @0xKurt ?